src/OrchardCore/OrchardCore.Users.Core/Services/UserStore.cs

```diff
@@ -201,9 +212,12 @@ public Task SetNormalizedUserNameAsync(IUser user, string normalizedName, Cancel
                 throw new ArgumentNullException(nameof(user));
             }
 
-            ((User)user).NormalizedUserName = normalizedName;
+            if (user is User su)
+            {
+                return Task.FromResult(su.NormalizedUserName);
+            }
 
-            return Task.CompletedTask;
+            return Task.FromResult<string>(null);
```

Comment:
And here should be

    return Task.CompletedTask;

I will make the change, hoping this is why the tests fails

Was it necessary to change this file?